plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -61,22 +64,36 @@ var defaultConfig = config.Configuration{
 	LatencyUnit:                 "ms",
 }
 
+func createTracingProvider() trace.TracerProvider {
+	exporter, err := stdout.New(stdout.WithPrettyPrint())
```

Comment:
We don't need tests printing traces. 
```suggestion
	exporter := tracetest.NewInMemoryExporter()
```
Add assertions to some tests to check that the spans were created correctly.